vms/platformvm/txs/executor/camino_tx_executor_test.go

```diff
@@ -547,3 +829,431 @@ func TestCaminoAddValidatorTxNodeSig(t *testing.T) {
+			expectedErr:              errInvalidSystemTxBody,
+		},
+		"Invalid outs (one excess)": {
+			ins:                      ins,
+			outs:                     append(outs, &avax.TransferableOutput{Out: &secp256k1fx.TransferOutput{}}),
+			preExecute:               func(t *testing.T, tx *txs.Tx) {},
+			generateUTXOsAfterReward: func(txID ids.ID) []*avax.UTXO { return UTXOsBeforeReward },
+			expectedErr:              errInvalidSystemTxBody,
+		},
+		"Invalid outs (wrong amount)": {
+			ins: ins,
+			outs: func() []*avax.TransferableOutput {
+				tempOuts := make([]*avax.TransferableOutput, len(outs))
+				copy(tempOuts, outs)
+				validOut := tempOuts[0].Out
+				if lockedOut, ok := validOut.(*locked.Out); ok {
+					validOut = lockedOut.TransferableOut
+				}
+				secpOut, ok := validOut.(*secp256k1fx.TransferOutput)
+				require.True(t, ok)
+
+				var invalidOut avax.TransferableOut = &secp256k1fx.TransferOutput{
+					Amt:          secpOut.Amt - 1,
+					OutputOwners: secpOut.OutputOwners,
+				}
+				if lockedOut, ok := validOut.(*locked.Out); ok {
+					invalidOut = &locked.Out{
+						IDs:             lockedOut.IDs,
+						TransferableOut: invalidOut,
+					}
+				}
+				tempOuts[0] = &avax.TransferableOutput{
+					Asset: avax.Asset{ID: env.ctx.AVAXAssetID},
+					Out:   invalidOut,
+				}
+				return tempOuts
+			}(),
+			preExecute:               func(t *testing.T, tx *txs.Tx) {},
+			generateUTXOsAfterReward: func(txID ids.ID) []*avax.UTXO { return UTXOsBeforeReward },
+			expectedErr:              errInvalidSystemTxBody,
+		},
+	}
+
+	execute := func(t *testing.T, tt test) (CaminoProposalTxExecutor, *txs.Tx) {
+		utx := &txs.CaminoRewardValidatorTx{
+			RewardValidatorTx: txs.RewardValidatorTx{TxID: stakerToRemove.TxID},
+			Ins:               tt.ins,
+			Outs:              tt.outs,
+		}
+
+		tx, err := txs.NewSigned(utx, txs.Codec, nil)
+		require.NoError(t, err)
+
+		tt.preExecute(t, tx)
+
+		onCommitState, err := state.NewDiff(lastAcceptedID, env)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		onAbortState, err := state.NewDiff(lastAcceptedID, env)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		txExecutor := CaminoProposalTxExecutor{
+			ProposalTxExecutor{
+				OnCommitState: onCommitState,
+				OnAbortState:  onAbortState,
+				Backend:       &env.backend,
+				Tx:            tx,
+			},
+		}
+		err = tx.Unsigned.Visit(&txExecutor)
+		require.ErrorIs(t, err, tt.expectedErr)
+		return txExecutor, tx
+	}
+
+	// Asserting UTXO changes
+	assertBalance := func(t *testing.T, tt test, tx *txs.Tx) {
+		env.state.SetHeight(uint64(1))
+		err := env.state.Commit()
+		require.NoError(t, err)
+
+		onCommitUTXOs, err := avax.GetAllUTXOs(env.state, outputOwners.AddressesSet())
+		require.NoError(t, err)
+		UTXOsAfterReward := tt.generateUTXOsAfterReward(tx.ID())
+		for i, onCommitUTXO := range onCommitUTXOs {
+			require.Equal(t, UTXOsAfterReward[i].TxID, onCommitUTXO.TxID)
```

Comment:
why not asserting whole utxo?
why not also asserting out index?

Reply:
Because the UTXO.id will be different. I ll not be able to know the generated ones and simulate the same.
What do you mean with index?

Reply:
UTXO contain UTXOID struct with txID and outputIndex, both of this fields used to create utxoID